ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/ManifestUtils.java

```diff
@@ -285,4 +303,96 @@ private static String splitOnComma(String value) {
 		sb.append(values[values.length - 1]);
 		return sb.toString();
 	}
+
+	/**
+	 * Returns the list all execution-environments required by the given OSGi
+	 * resource. Only registered EEs are considered.
+	 *
+	 * @param resource
+	 *            the osgi resource
+	 * @return a list containing the id's of all required EEs
+	 * @see ExecutionEnvironmentNamespace#EXECUTION_ENVIRONMENT_NAMESPACE
+	 * @see IExecutionEnvironmentsManager#getExecutionEnvironments()
+	 */
+	public static List<String> getRequiredExecutionEnvironments(Resource resource) {
+		if (resource instanceof BundleDescription bundleDescription) {
+			// fast path if a BREE header is specified in the MANIFEST.MF
+			String[] requiredEE = bundleDescription.getExecutionEnvironments();
+			if (requiredEE.length > 0) {
+				return List.of(requiredEE);
+			}
+		}
+		List<Requirement> requirements = resource.getRequirements(EXECUTION_ENVIRONMENT_NAMESPACE);
+		return requirements.stream()
+				.map(requirement -> requirement.getDirectives().get(Namespace.REQUIREMENT_FILTER_DIRECTIVE))
+				.filter(Objects::nonNull) //
+				.mapMulti(ManifestUtils::parseRequiredEEsFromFilter) //
+				.toList();
+	}
+
+	// provide fast-path for simple filters like: (&(osgi.ee=JavaSE)(version=17)
+	private static final Map<String, String> SIMPLE_EE_FILTERS = new HashMap<>();
+	private static final Map<String, Map<String, String>> AVAILABLE_EE_ATTRIBUTES = new HashMap<>();
+
+	static {
+		IExecutionEnvironment[] availableEEs = JavaRuntime.getExecutionEnvironmentsManager().getExecutionEnvironments();
+		for (IExecutionEnvironment ee : availableEEs) {
+			String id = ee.getId();
+			String eeName;
+			String eeVersion;
+			// See OSGi spec chapter 3.4.1 Bundle-RequiredExecutionEnvironment
+			if (id.indexOf('/') < id.indexOf('-')) {
+				int versionSeparator = id.lastIndexOf('-');
+				if (versionSeparator < 0) {
+					throw new IllegalArgumentException("Missing version-separator in EE Id"); //$NON-NLS-1$
+				}
+				eeName = id.substring(0, versionSeparator);
+				if ("J2SE".equals(eeName)) { //$NON-NLS-1$
+					eeName = "JavaSE"; //$NON-NLS-1$
+				}
+				eeVersion = id.substring(versionSeparator + 1);
+			} else {
+				try {
+					Method getOSGiEENameVersion = OSGiManifestBuilderFactory.class
+							.getDeclaredMethod("getOSGiEENameVersion", String.class); //$NON-NLS-1$
```

Comment:
@tjwatson can this be made available to PDE in some way since I would like to avoid to duplicate the logic in PDE.
It does not have to be an API, some internals that are accessible should be fine.

Reply:
In its current form it would have to be internal since returning a String[] is rather strange.  Would something like this help as possible API:

```java
public static Map<String, String> getOSGiEERequirementDirectives(String bree)
```
Where the directives just have the osgi.ee `filter` directive?
Or maybe just a method to create the filter:

```java
public static String getOSGiEERequirementFilter(String bree)
```